### DISCUSSION TITLE
Password History

### DISCUSSION THREAD

**Discussion by masterwishx** [dani-garcia/vaultwarden#5141] [Q&A] [answered]
Do we have setting for password history ?
cant find it , what default value ?

**Comment by stefan0xC** [comment]
Passwort history is enabled by default so there is nothing to configure.

**Reply by masterwishx** [reply]
> Passwort history is enabled by default so there is nothing to configure.

How much passwords we can have in history?

**Comment by BlackDex** [comment]
https://bitwarden.com/help/product-faqs/#q-i-need-an-old-password-can-i-view-the-history-of-a-password-that-i-changed-in-bitwarden

**Reply by masterwishx** [reply]
Thanks, i meaned if we can set/increase  the count of passwords in history . how much we have by default?

**Reply by BlackDex** [reply]
No. The clients only send 5 historical items. Vaultwarden doesn't have a limit.

**Reply by masterwishx** [reply]
> The clients only send 5 historical items. Vaultwarden doesn't have a limit.

Do you mean only 5 password in history for login or no limit ?
What about password generator history, also no limit ?

**Reply by BlackDex** [reply, marked as answer]
The Bitwarden clients only return 5 historical passwords on save. Vaultwarden stores this as a blob without looking at it.
The password generator is stored client side, nothing is send to the server regarding that.
